lib/core/Observer.ts

```diff
@@ -104,13 +111,26 @@ export default class Observer {
 
         // Queue parallel downloading and processing of batch files.
         for (const transaction of transactions) {
-          const awaitingTransaction = {
-            transaction: transaction,
-            processingStatus: TransactionProcessingStatus.Pending
-          };
-          this.transactionsUnderProcessing.push(awaitingTransaction);
-          // Intentionally not awaiting on downloading and processing each operation batch.
-          void this.processTransaction(transaction, awaitingTransaction);
+          if (transaction.transactionTime === this.currentBlockHeight) {
+            this.transactionsInCurrentBlock.push(transaction);
+          } else {
+            if (this.currentBlockHeight !== undefined) {
+              const throughputLimiter = this.versionManager.getThroughputLimiter(this.currentBlockHeight);
+              const qualifiedTransactionsInCurrentBlock = await throughputLimiter.selectQualifiedTransactions(this.transactionsInCurrentBlock);
```

Comment:
We also need to update `protocol.md` describing rate/throughput limiting as one of the  anti-spam mechanism, under DDoS Attack & Mitigation section, and I am fine if it comes in as a separate PR.